test/com/amazon/ion/impl/bin/WriteBufferTest.java

```diff
@@ -959,4 +959,118 @@ public void shiftBytesLeftAcrossBufferBlocksExclusively() throws IOException {
         buf.shiftBytesLeft(5, 5);
         assertBuffer("012345BCDEF".getBytes());
     }
+
+    @Test
+    public void shiftBytesLeftAcrossBufferBlocksEmptyingLastBlock() throws IOException {
+        assertEquals(11, ALLOCATOR.getBlockSize());
+        // The "B" is the first and only byte in the second block.
+        // Shifting 1 byte left by one empties the last block.
+        buf.writeBytes("0123456789AB".getBytes());
+        buf.shiftBytesLeft(1, 1);
+        assertBuffer("0123456789B".getBytes());
+    }
+
+    @Test
+    public void shiftEntireBlock() throws IOException {
+        assertEquals(11, ALLOCATOR.getBlockSize());
+        // The buffer contains two full blocks
+        buf.writeBytes("0123456789|ABCDEFGHIJ|".getBytes());
+        // We shift an entire block left by the block size
+        buf.shiftBytesLeft(11, 11);
+        assertBuffer("ABCDEFGHIJ|".getBytes());
+    }
+
+    @Test
+    public void shiftBytesLeftByMoreThanTheBlockSize() {
+        assertEquals(11, ALLOCATOR.getBlockSize());
+        // We have 5 blocks' worth of data
+        buf.writeBytes("0123456789|0123456789|0123456789|0123456789|0123456789|".getBytes());
+        // We can shift left amounts greater than the block size
+        buf.shiftBytesLeft(24, 24);
+        buf.writeBytes("0123456789|0123456789|012345678".getBytes());
```

Comment:
I spent quite a while step-through debugging this test, because this assertion didn't look right. But the code was working perfectly...

Then I realized there's a reason this assertion wasn't failing... because it's not an assertion! The following assertion works:

```
assertBuffer("01234569|0123456789|0123456789|".getBytes());
```

Reply:
🤦 I really should've slept on this commit. Don't code drowsy, kids.